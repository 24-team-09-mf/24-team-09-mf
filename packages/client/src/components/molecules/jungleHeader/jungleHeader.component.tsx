import { Wrapper, Image, Title } from './jungleHeader.styles'

import jungleTileset from '@/assets/images/landing/jungle_tileset.png'

const VARIANTS = {
  xs: 'xs',
  sm: 'sm',
  md: 'md',
  lg: 'lg',
  xl: 'xl',
}

const VARIANT_FONT_SIZES = {
  xs: '16px',
  sm: '24px',
  md: '32px',
  lg: '42px',
  xl: '48px',
}

const ALIGN_VARIANTS = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
}

interface IJungleHeaderProps {
  title: string
  variant?: keyof typeof VARIANTS
  align?: 'left' | 'center' | 'right'
}

const JungleHeader: React.FC<IJungleHeaderProps> = ({
  title,
  variant = 'xl',
  align = 'center',
}) => {
  const fontSize = VARIANT_FONT_SIZES[variant]
    ? VARIANT_FONT_SIZES[variant]
    : VARIANT_FONT_SIZES['xl']

  return (
    <Wrapper align={ALIGN_VARIANTS[align]}>
      <Image src={jungleTileset} size={fontSize} />
      <Title fontSize={fontSize}>{title}</Title>
    </Wrapper>
  )
}

export default JungleHeader
