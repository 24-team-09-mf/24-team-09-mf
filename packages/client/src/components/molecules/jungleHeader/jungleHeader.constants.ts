export const VARIANTS = {
  xs: 'xs',
  sm: 'sm',
  md: 'md',
  lg: 'lg',
  xl: 'xl',
}

export const VARIANT_FONT_SIZES = {
  xs: '16px',
  sm: '24px',
  md: '32px',
  lg: '42px',
  xl: '48px',
}

export const ALIGN_VARIANTS = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
}
