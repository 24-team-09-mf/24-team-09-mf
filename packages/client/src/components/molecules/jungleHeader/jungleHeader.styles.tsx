import styled from 'styled-components'

export const Wrapper = styled.div((props: { align: string }) => {
  return `
  display: flex;
  align-items: center;
  justify-content: ${props.align};
  margin: 20px 0;
`
})

export const Image = styled.img((props: { size: string }) => {
  return `
    height: ${props.size};
    width: auto;
    margin-right: 1.5rem;
`
})

export const Title = styled.h2((props: { fontSize: string }) => {
  return `
    font-size: ${props.fontSize};
    font-weight: bold;
    text-transform: uppercase;
    line-height: 100%;
  `
})
