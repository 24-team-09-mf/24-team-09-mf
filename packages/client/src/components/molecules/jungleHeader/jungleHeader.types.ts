import { VARIANTS } from './jungleHeader.constants'

export interface IJungleHeaderProps {
  title: string
  variant?: keyof typeof VARIANTS
  align?: 'left' | 'center' | 'right'
}
