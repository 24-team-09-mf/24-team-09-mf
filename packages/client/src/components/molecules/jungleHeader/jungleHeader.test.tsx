import * as renderer from 'react-test-renderer'
import { JungleHeader } from './jungleHeader.component'
describe('Тест компонента ErrorBlock', () => {
  it('Тестирование снепшотом', () => {
    const component = renderer
      .create(<JungleHeader title="hello world" />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
