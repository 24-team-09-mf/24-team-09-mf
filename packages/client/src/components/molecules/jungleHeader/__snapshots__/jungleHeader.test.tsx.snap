// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента ErrorBlock Тестирование снепшотом 1`] = `
<div
  className="sc-beqWaB HcakY"
>
  <img
    className="sc-gueYoa eJXQMp"
    size="48px"
  />
  <h2
    className="sc-dmqHEX ggbouM"
    fontSize="48px"
  >
    hello world
  </h2>
</div>
`;
